### PR TITLE
[LoongArch64] amend the crossgen2 for LA64.

### PR DESCRIPTION
**Pull request by shushanhf**
This PR is part of the issue https://github.com/dotnet/runtime/issues/69705 to amend the LA's port.

Amend the crossgen2 for LA64.

### CODE REVIEW

**Comment by shushanhf** [comment]
The failed tests are `Timed Out`
```
  nativeaot/SmokeTests/SharedLibrary/SharedLibrary/SharedLibrary.sh [FAIL]
        
        cmdLine:/Users/runner/work/1/s/artifacts/tests/coreclr/OSX.x64.Release/nativeaot/SmokeTests/SharedLibrary/SharedLibrary/SharedLibrary.sh Timed Out (timeout in milliseconds: 1800000 from variable __TestTimeout, start: 9/6/2022 1:24:23 PM, end: 9/6/2022 1:54:24 PM)
        
        Return code:      -100
        Raw output file:      /Users/runner/work/1/s/artifacts/tests/coreclr/OSX.x64.Release/nativeaot/SmokeTests/Reports/nativeaot.SmokeTests/SharedLibrary/SharedLibrary/SharedLibrary.output.txt
        Raw output:

```

Hi, @jkotas 
Could you please review this PR ?

File: src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TransitionBlock.cs [hunk 1]
```diff
@@ -387,8 +387,8 @@ public void ComputeReturnValueTreatment(CorElementType type, TypeHandle thRetTyp
                         }
                     }
 
-                    // Value types are returned using return buffer by default
-                    usesRetBuffer = true;
+                    // Value types are returned using return buffer by default except LoongArch64.
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TransitionBlock.cs:390]: This should match the logic in https://github.com/dotnet/runtime/blob/a9b18fc97a700f2012ced4042b470fefaf6af313/src/coreclr/vm/callingconvention.h#L1886-L1887 .

Comment (shushanhf) [reply src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TransitionBlock.cs:390]: Thanks

Comment (shushanhf) [reply src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TransitionBlock.cs:390]: Hi, @jkotas 
Are these `RETURN_HAS_RET_BUFFER` and `usesRetBuffer`  liking the X8_return_buffer for ARM64 ?

If these are not liking the ARM64's X8_return_buffer, there is no need to modify and I will revert the `usesRetBuffer = !IsLoongArch64`.
As the logic 
https://github.com/dotnet/runtime/blob/a9b18fc97a700f2012ced4042b470fefaf6af313/src/coreclr/vm/callingconvention.h#L1779-L1786
where it is only the TYPEDBYREF.

Comment (shushanhf) [reply src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TransitionBlock.cs]: Thanks
I had update it.

**Review by jkotas — APPROVED** [review]

**Pull request merged**